gazebo_plugins/test/test_gazebo_ros_ray_sensor.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <gazebo_ros/testing_utils.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/range.hpp>
+#include <sensor_msgs/msg/point_cloud.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <sensor_msgs/msg/laser_scan.hpp>
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include <atomic>
+
+/// Tests the gazebo_ros_ray_sensor plugin
+class TestGazeboRosRaySensor : public ::testing::Test
+{
+public:
+  TestGazeboRosRaySensor() {}
+
+  // Documentation inherited
+  static void SetUpTestCase();
+  // Documentation inherited
+  static void TearDownTestCase();
+
+protected:
+  /// command line arguments to pass to gzserver
+  static const std::vector<const char *> args;
+  /// Wrapper for gzserver run with plugin
+  static std::unique_ptr<gazebo_ros::GazeboProcess> gazebo_process_;
+  /// ROS node used to verify output of plugin
+  static rclcpp::Node::SharedPtr node_;
+  /// ROS executor to call node_'s callbacks
+  static rclcpp::executor::Executor::SharedPtr executor_;
```

Comment:
Just curious, why are you making these static?

Reply:
Rewrote test to actually verify correctness of output using gazebo's ground truth in 
3f117fe